### PR TITLE
SCMOD-7632: Update liquibase installer logging

### PR DESCRIPTION
**Pull request by gaana-vishwnath**
JENKINS: http://sou-jenkins2.hpeswlab.net/job/CAFapi/job/CAFapi~caf-common~SCMOD-7632-GV~CI/2/console

JIRA: https://portal.digitalsafe.net/browse/SCMOD-7632

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,10 +57,10 @@
         <module>swagger-ui</module>
         <module>util-ref</module>
         <module>util-jerseycompat</module>
+        <module>util-process-identifier</module>
         <module>util-liquibase-installer</module>
         <module>util-moduleloader</module>
-        <module>util-naming</module>
-        <module>util-process-identifier</module>
+        <module>util-naming</module>
```
Comment (dermot-hardy) [inline pom.xml]: Whitespace

File: util-liquibase-installer/pom.xml [hunk 2]
```diff
@@ -46,11 +46,22 @@
         <dependency>
             <groupId>org.springframework</groupId>
             <artifactId>spring-core</artifactId>
+            <exclusions>
+                <exclusion>
+                    <artifactId>commons-logging</artifactId>
+                    <groupId>commons-logging</groupId>
+                </exclusion>
+            </exclusions>
```
Comment (dermot-hardy) [inline util-liquibase-installer/pom.xml]: Remove from here since this should be covered at the dependency management level.

File: util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java [hunk 3]
```diff
@@ -84,14 +82,13 @@ private void run(final String[] args) throws SQLException, LiquibaseException
...
-            System.err.println("java Main [options...] arguments...");
+            logger.error(e.getMessage());
+            logger.error("java Main [options...] arguments...");
             // print the list of available options
             parser.printUsage(System.err);
-            System.err.println();
 
             // print option sample. This is useful some time
-            System.err.println(
+            logger.error(
```
Comment (dermot-hardy) [inline util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java]: Leave this function pointing to `System.err` (since we don't want to diverge from `parser.printUsage`).

File: util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java [hunk 4]
```diff
@@ -179,23 +174,25 @@ private void updateDB() throws SQLException, LiquibaseException
...
                         String errorMessage = "No liquibase changelog-master.xml or changelog.xml could be located";
-                        Logger.getLogger(Application.class.getName()).log(Level.SEVERE, errorMessage);
+                        logger.error(errorMessage);
                         throw new RuntimeException(errorMessage);
                     }
-                    liquibase.getLog().setLogLevel(logLevel);
-                    liquibase.update(new Contexts());
-                } catch (final IOException ioe) {
-                    Logger.getLogger(Application.class.getName()).log(Level.SEVERE, ioe.getMessage(), ioe);
+                } catch (IOException ex) {
```
Comment (dermot-hardy) [inline util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java]: `final`

File: util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java [hunk 5]
```diff
@@ -179,23 +174,25 @@ private void updateDB() throws SQLException, LiquibaseException
...
+                    logger.error(ex.getMessage());
                 }
+                
+                liquibase.update(new Contexts());
```
Comment (dermot-hardy) [inline util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java]: Put back inside the catch block.

File: util-moduleloader/pom.xml [hunk 6]
```diff
@@ -35,6 +35,11 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.github.cafapi</groupId>
+            <artifactId>caf-logging</artifactId>
+            <scope>runtime</scope>
+        </dependency>
```
Comment (dermot-hardy) [inline util-moduleloader/pom.xml]: Remove this.  Only the outermost component should specify the logging framework.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (gaana-vishwnath) [reply pom.xml]: corrected

File: util-liquibase-installer/pom.xml (hunk 2 above)
Comment (gaana-vishwnath) [reply util-liquibase-installer/pom.xml]: removed

File: util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java (hunk 4 above)
Comment (gaana-vishwnath) [reply util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java]: corrected

File: util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java (hunk 3 above)
Comment (gaana-vishwnath) [reply util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java]: added it back

File: util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java (hunk 5 above)
Comment (gaana-vishwnath) [reply util-liquibase-installer/src/main/java/com/hpe/caf/utilliquibaseinstaller/Application.java]: corrected

File: util-moduleloader/pom.xml (hunk 6 above)
Comment (gaana-vishwnath) [reply util-moduleloader/pom.xml]: removed

**Comment by gaana-vishwnath** [comment]
updated as per review comments. Kindly review again.

File: pom.xml [hunk 7]
```diff
@@ -57,10 +57,10 @@
         <module>swagger-ui</module>
         <module>util-ref</module>
         <module>util-jerseycompat</module>
+        <module>util-process-identifier</module>
         <module>util-liquibase-installer</module>
         <module>util-moduleloader</module>
         <module>util-naming</module>
-        <module>util-process-identifier</module>
```
Comment (dermot-hardy) [inline pom.xml:63]: Is this change necessary?  Does this have to move location?

File: pom.xml [hunk 8]
```diff
@@ -136,6 +137,11 @@
                 <artifactId>caf-api</artifactId>
                 <version>1.17.0-SNAPSHOT</version>
             </dependency>
+            <dependency>
+                <groupId>com.github.cafapi</groupId>
+                <artifactId>caf-logging</artifactId>
+                <version>1.0.0-SNAPSHOT</version>
+            </dependency>
```
Comment (dermot-hardy) [inline pom.xml:144]: This has been released now, and note that the Maven coordinates have changed.  It should now be:

    <groupId>com.github.cafapi.logging</groupId>
    <artifactId>caf-logging-logback</artifactId>
    <version>1.0.0-36</version>

File: util-liquibase-installer/pom.xml [hunk 9]
```diff
@@ -51,6 +51,11 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-beans</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.github.cafapi</groupId>
+            <artifactId>caf-logging</artifactId>
```
Comment (dermot-hardy) [inline util-liquibase-installer/pom.xml:56]: Maven coordinates changed.

**Review by dermot-hardy — APPROVED** [review]
